wordpress-mobile/WordPress-Aztec-iOS
# Improve Mark formatting

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6517
- https://github.com/WordPress/gutenberg/pull/57650
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22425

This PR fixes several issues related to the Mark formatting.

- It completes the missing functionality implementation to fully support the Mark tag.
- Adds `MarkStringAttributeConverter` which will handle toggling the format as well as returning the expected HTML format to Gutenberg, by removing the extra spaces. This was causing issues like history changes or showing the active style indicator when it shouldn't.
- Updates `MarkFormatter` to set the expected CSS attributes as well as adding a `color` param.  
- Adds `preprocessMarkForInsertion` that will handle cases like auto-corrected strings, it will check if the new auto-corrected words need to have the Mark representation along with `adjustAttributesForMark`.
- It also fixes `toggleMark` by allowing passing a `color` param and to be able to change colors while editing a Mark formatting or resetting the existing format.

## To test

Check this [PR description](https://github.com/WordPress/gutenberg/pull/57650) under the `Test` section. Where there are builds linked as well.
